src/vs/editor/contrib/linesOperations/linesOperations.ts

```diff
@@ -672,146 +679,132 @@ export class JoinLinesAction extends EditorAction {
-				}
+	private _getLinesToJoin(editor: IActiveCodeEditor): IJoinLinesOperation[] {
+		// Construct Join operations
+		let operations: IJoinLinesOperation[] = editor.getSelections().map((s) => {
 
-				if (currentValue.startLineNumber > previousValue.endLineNumber + 1) {
-					reducedSelections.push(previousValue);
-					return currentValue;
-				} else {
-					return new Selection(previousValue.startLineNumber, previousValue.startColumn, currentValue.endLineNumber, currentValue.endColumn);
-				}
-			} else {
-				if (currentValue.startLineNumber > previousValue.endLineNumber) {
-					reducedSelections.push(previousValue);
-					return currentValue;
-				} else {
-					return new Selection(previousValue.startLineNumber, previousValue.startColumn, currentValue.endLineNumber, currentValue.endColumn);
-				}
+			let endLineNumber = s.endLineNumber;
+			if (s.startLineNumber < s.endLineNumber && s.endColumn === 1) {
+				endLineNumber -= 1;
 			}
+
+			return {
+				startLineNumber: s.startLineNumber,
+				selectionStartColumn: s.selectionStartColumn,
+				endLineNumber: endLineNumber,
+				positionColumn: s.positionColumn
+			};
 		});
 
-		reducedSelections.push(lastSelection);
+		// Sort Join operations
+		operations.sort((a, b) => {
+			if (a.startLineNumber === b.startLineNumber) {
+				return a.endLineNumber - b.endLineNumber;
+			}
+			return a.startLineNumber - b.startLineNumber;
+		});
 
-		let model = editor.getModel();
-		if (model === null) {
-			return;
+		// Merge Join operations which are adjacent or overlapping
+		let mergedOperations: IJoinLinesOperation[] = [];
+		let previousOperation = operations[0];
+		for (let i = 1; i < operations.length; i++) {
+			if (previousOperation.endLineNumber + 1 >= operations[i].startLineNumber) {
+				// Merge current operations into the previous one
+				previousOperation.endLineNumber = operations[i].endLineNumber;
+			} else {
+				// Push previous operation
+				mergedOperations.push(previousOperation);
+				previousOperation = operations[i];
+			}
 		}
+		// Push the last operation
+		mergedOperations.push(previousOperation);
 
-		let edits: IIdentifiedSingleEditOperation[] = [];
-		let endCursorState: Selection[] = [];
-		let endPrimaryCursor = primaryCursor;
-		let lineOffset = 0;
+		return mergedOperations;
+	}
 
-		for (let i = 0, len = reducedSelections.length; i < len; i++) {
-			let selection = reducedSelections[i];
-			let startLineNumber = selection.startLineNumber;
-			let startColumn = 1;
-			let columnDeltaOffset = 0;
-			let endLineNumber: number,
-				endColumn: number;
+	public run(_accessor: ServicesAccessor, editor: ICodeEditor): void {
+		if (!editor.hasModel()) {
+			return;
+		}
+		let model = editor.getModel();
 
-			let selectionEndPositionOffset = model.getLineContent(selection.endLineNumber).length - selection.endColumn;
+		let removeLines = this._getLinesToJoin(editor);
+		let ops: IIdentifiedSingleEditOperation[] = [];
+		let conString: String[] = [];
```

Comment:
When shortening names it makes it harder to understand what it is used for?  What is con?  Would naming the variable more clearly make it more readable?